pkg/base/variadic.go

```diff
@@ -0,0 +1,339 @@
+package base
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	kotsv1beta1 "github.com/replicatedhq/kots/kotskinds/apis/kots/v1beta1"
+	upstreamtypes "github.com/replicatedhq/kots/pkg/upstream/types"
+	yaml3 "gopkg.in/yaml.v3"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+// Known issues and TODOs:
+// This currently only addresses variadic items.  Variadic groups are not included yet and may require changes to these functions.
+// Test_renderReplicated may sometimes fail because the resulting array can be out of order.  A more specific results check could be executed instead.
+// getVariadicGroupsForTemplate should be split into subfunctions to make it easier to read
+// The last element in the YamlPath must be an array
+
+func processVariadicConfig(u *upstreamtypes.UpstreamFile, config *kotsv1beta1.Config) error {
```

Comment:
Looks like this function will panic if upstream file is not a yaml file, like a helm chart.

Reply:
added some safety handling here.  Files that cannot be unmarshaled as yaml, do not have valid `apiVersion` or `kind` fields, or have a name or namespace that is not a string will be skipped for variadic processing.  These files would not match any variadic template targets if we continued processing them anyway.